crates/bevy_ecs/src/component/type_info.rs

```diff
@@ -0,0 +1,67 @@
+use std::{alloc::Layout, any::TypeId};
+
+/// Metadata required to store a component
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TypeInfo {
+    id: TypeId,
```

Comment:
I think the following is a bit nicer, but no strong preference.
```suggestion
    type_id: TypeId,
```

Reply:
agreed!